src/webgpu/gpu_test.ts

```diff
@@ -846,13 +854,26 @@ export class GPUTestBase extends Fixture<GPUTestSubcaseBatchState> {
       ${textureSrcCode}
       @group(0) @binding(1) var<storage, read_write> dst : Buffer;
 
-      @compute @workgroup_size(1) fn main() {
-        var coord = vec2<i32>(0, 0);
-        for (var sampleIndex = 0; sampleIndex < ${sampleCount};
+      struct Params {
+        origin: vec3u,
```

Comment:
nit: If only supporting 2d textures, I think use vec2u instead of vec3u+pad. We can add vec3u later if we extend this to 3d. Right now it makes it a bit harder to read.